File: 2. Structural Design Patterns/Flyweight/Python/FlyweightTemplate.py
# Flyweight Pattern – Python Template

class Flyweight:
    def __init__(self, intrinsic_state):
        self.intrinsic_state = intrinsic_state
    
    def operation(self, extrinsic_state):
        return f"Flyweight: intrinsic={self.intrinsic_state}, extrinsic={extrinsic_state}"

class FlyweightFactory:
    _flyweights = {}
    
    @classmethod
    def get_flyweight(cls, key):
        if key not in cls._flyweights:
            cls._flyweights[key] = Flyweight(key)
        return cls._flyweights[key]

class Context:
    def __init__(self, key, extrinsic_state):
        self.flyweight = FlyweightFactory.get_flyweight(key)
        self.extrinsic_state = extrinsic_state
    
    def operation(self):
        return self.flyweight.operation(self.extrinsic_state)

def main():
    context1 = Context("shared_key", "unique_state_1")
    context2 = Context("shared_key", "unique_state_2")
    print(context1.operation())
    print(context2.operation())

if __name__ == "__main__":
    main()

File: 2. Structural Design Patterns/Flyweight/Go/FlyweightTemplate.go
package main

// Flyweight Pattern – Go Template

type Flyweight struct{ /* intrinsic, immutable data */ }

type Factory struct{}
var fwCache = map[string]Flyweight{}

func (Factory) Get(key string) Flyweight {
	if fw, ok := fwCache[key]; ok { return fw }
	fw := Flyweight{}
	fwCache[key] = fw
	return fw
}

type Context struct{ fw Flyweight /* extrinsic fields here */ }

func NewContext(key string) Context { return Context{ fw: Factory{}.Get(key) } }
func (c Context) Operation(/* extrinsic params */) { /* use c.fw + extrinsic */ }

func main(){ c := NewContext("key"); c.Operation() }
